Zend/zend_builtin_functions.c

```diff
@@ -635,14 +627,14 @@ ZEND_FUNCTION(get_called_class)
    Retrieves the parent class name for object or class or current scope or false if not in a scope. */
 ZEND_FUNCTION(get_parent_class)
 {
-	zval *arg;
+	zval *arg = NULL;
 	zend_class_entry *ce = NULL;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|z", &arg) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|z!", &arg) == FAILURE) {
```

Comment:
There's https://wiki.php.net/rfc/get_class_disallow_null_parameter which dropped the `!` from get_class...

Reply:
I guess it's not very nice to reintroduce the support for a null parameter value in `get_class()` so that it's consistent with `get_parent_class()` after my changes. Or is it? Then what about deprecating to call these function with no parameters as part of the PHP 8 deprecations RFC? Now, that we always have `$this`, we can just call `get_*_class($this)`.

Reply:
I've added it a while bakc: https://wiki.php.net/rfc/deprecations_php_8_0

Reply:
Saviour! :) Please don't forget to add `get_parent_class()`, too :)

Reply:
I've added it as a separate discussion, as it seems to me that the case for ``get_parent_class()`` is less strong than for ``get_class()``, feel free to edit this one in the RFC for a better rationale